runtime/lua/vim/net.lua

```diff
@@ -0,0 +1,71 @@
+local M = {}
+
+--- Makes an HTTP GET request to the given URL.
+---
+--- This function operates in two modes:
+---   - Synchronous (blocking): If `on_response` is omitted, it waits for the
+---   request to complete and returns the result.
+---   - Asynchronous (non-blocking): If `on_response` is provided, it returns
+---   immediately and invokes the callback with the result later.
+---
+--- @param url string The URL for the request.
+--- @param opts? table Optional parameters:
+---   - `verbose` (boolean|nil): Enables curl verbose output.
+---   - `retry`   (integer|nil): Number of retries on transient failures (default: 3).
+---   - `output`  (string|nil): A file path to save the response body to. If set, the success value will be `true` instead of the response body.
+--- @param on_response? fun(err?: string, content?: string|boolean) Optional callback for async execution.
+---   It is invoked with `(err, nil)` on failure or `(nil, content)` on success.
+--- @return string|boolean|nil In sync mode, returns the response body or `true` on success; otherwise `nil`
```

Comment:
the wording is kind of confusing, does this make sense:
```suggestion
--- @return string|boolean|nil Response body (sync mode) or `true` (async) on success; `nil` on failure
```